cfme/base/__init__.py

```diff
@@ -46,6 +47,25 @@ def name(self):
         # empty string default for string building w/o None
         return getattr(self.appliance._rest_api_server(), 'name', '')
 
+    @property
+    def tree_path(self):
+        """Generate the path list for navigation purposes
+        list elements follow the tree path in the configuration accordion
+
+        Returns:
+            list of path elements for tree navigation
+        """
+        current = '' if self.is_slave else ' (current)'
+        # replace two spaces with one, formatting tweak when name is empty
+        name_string = f' {self.name} '.replace('  ', ' ')
```

Comment:
Can you tell me more about what are you doing here ? Not sure if I am familiar with what this property does and what's up with the spaces?

Reply:
When the name attribute is an empty string, you end up with two spaces. Trying to format this into the complete string in which its used, there has to be leading/following spaces, but only one space when the name is empty.  This just injects the name, and then collapses two spaces to one if the name was empty - without checking what self.name is and having a ternary.